mopidy/mpd/__init__.py

```diff
@@ -19,7 +19,7 @@ def get_default_config(self):
     def get_config_schema(self):
         schema = super(Extension, self).get_config_schema()
         schema['hostname'] = config.Hostname()
-        schema['port'] = config.Port()
+        schema['port'] = config.Port(optional=True)
```

Comment:
So I tried out the following for `Hostname.deserialize` to try and reuse the existing Path validation code. What do you think?
```
    def deserialize(self, value, display=False):
        validators.validate_required(value, self._required)
        if not value.strip():
            return None
        socket_path = path.get_unix_socket_path(value)
        if socket_path is not None:
            return Path(not self._required).deserialize(socket_path)
        try:
            socket.getaddrinfo(value, None)
        except socket.error:
            raise ValueError('must be a resolveable hostname or valid IP')
        return value
```
I also moved that regex code to a `path.get_unix_socket_path()` helper since it occurs in the `internal.network` code too.

Reply:
Yeah okay that's much more modular. So just the regex match and group extraction is contained in that helper method? Then delegating path expansion and validation to the Path class?